src/main/java/seedu/address/model/student/CourseContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,33 @@
+package seedu.address.model.student;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Course}'s {@code courseCode} matches any of the keywords given.
+ */
+public class CourseContainsKeywordsPredicate implements Predicate<Student> {
+    private final List<String> keywords;
+
+    public CourseContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Student student) {
+        return keywords.stream()
+                .anyMatch(keyword ->
+                        student.getCourses().stream()
```

Comment:
I understand this works, but maybe for the sake of readability / slap / less nesting etc, maybe a separate function to get a student's courses as a `List<String>` could be used? (Then we can do `.anyMatch` here.)